Plugins/MatrixPlugin/MatrixForm.cs

```diff
@@ -0,0 +1,257 @@
+                lDefendSkillName.DoDataBinding("Text", logic, "currentAction.DefenceSkill");
+
+                DoDataBindingWithFormatter(lActionAttributeValue, "currentAction.ActionAttribute", new ConvertEventHandler(AttributeToValue));
+                DoDataBindingWithFormatter(lActionSkillValue, "currentAction.ActionSkill", new ConvertEventHandler(SkillToValue));
+                DoDataBindingWithFormatter(lSkillLimitValue, "currentAction.Limit", new ConvertEventHandler(MatrixAttributeToValue));
+                DoDataBindingWithFormatter(lDefendAttributeValue, "currentAction.DefenceAttribute", new ConvertEventHandler(AttributeToValue));
+                DoDataBindingWithFormatter(lDefendSkillValue, "currentAction.DefenceSkill", new ConvertEventHandler(MatrixAttributeToValue));
+                lActionModifier.DoDataBinding("Text", logic, "currentAction.ActionModifier");
+                lDefendModifier.DoDataBinding("Text", logic, "currentAction.DefenceModifier");
+                dpcActionDicePool.DoDataBinding("DicePool", logic, "ActionDicePool");
+                dpcDefendDicePool.DoDataBinding("DicePool", logic, "DefenceDicePool");
+            }
+        }
+
+
+        private void DoDataBindingWithFormatter(Control obj, string dataMember, ConvertEventHandler formatter)
+        {
+            Binding binding = new Binding("Text", logic, dataMember);
+            binding.Format += formatter;
+            obj.DataBindings.Add(binding);
+        }
+
+        private void AttributeToValue(object sender, ConvertEventArgs cevent)
+        {
+            if (cevent.DesiredType != typeof(string)) return;
+            cevent.Value = logic.GetTotalAttribute((string)cevent.Value).ToString();
+        }
+
+        private void MatrixAttributeToValue(object sender, ConvertEventArgs cevent)
+        {
+            if (cevent.DesiredType != typeof(string)) return;
+            cevent.Value = logic.GetTotalMatrixAttribute((string)cevent.Value).ToString();
+        }
+        private void SkillToValue(object sender, ConvertEventArgs cevent)
+        {
+            if (cevent.DesiredType != typeof(string)) return;
+            cevent.Value = logic.GetTotalSkill((string)cevent.Value).ToString();
+        }
+
+        private void AddRadioCheckedBinding(RadioButton radio, object dataSource, string dataMember, string trueValue)
+        {
+            var binding = new Binding(nameof(RadioButton.Checked), dataSource, dataMember, true, DataSourceUpdateMode.OnPropertyChanged);
+            binding.Parse += (s, a) => { if ((bool)a.Value) a.Value = trueValue; logic.OverClocked = trueValue; };
+            binding.Format += (s, a) => a.Value = (a.Value).Equals(trueValue);
+            radio.DataBindings.Add(binding);
+        }
+
+        public void RefreshMatrixAttributeCBOs(MatrixLogic objThis, ComboBox cboAttack, ComboBox cboSleaze, ComboBox cboDP, ComboBox cboFirewall)
+        {
+            if (objThis == null)
```

Comment:
Wait, what? You're already storing a `MatrixLogic` object inside this class that is meant to be this class' main thing. Why are you now including a `MatrixLogic` argument in this method, meaning that this method can be called with a `MatrixLogic` object that is not the same `MatrixLogic` as this class' private `logic`?

Reply:
edit it in https://github.com/chummer5a/chummer5a/pull/4563/commits/0f130c19bc6cec27e978019f38ca192c428d1f3a